### DISCUSSION TITLE
const { data, error } = await supabase.auth.admin.getUserByEmail(invitedUserEmail)

### DISCUSSION THREAD

**Discussion by MichaelJCole** [supabase/supabase#31186] [Questions]
Hi, I have an "invite user by email" feature to implement.  

How can I get a user's id from their email?   I want this for my "invites" table where users can accept/reject their invites in the app.  Clicking the email link does not automatically accept the invite.  My user flows are:

Invite new user:
- createUser()
- add userid to invites table
- email password reset link
- password reset form forwards to their dashboard where they accept/reject their invites.
- POST api/invite/respond authenticates the already logged in user.

Invite an existing user:
- findUserByEmail()
- add userid to invites table
- email link to the app dashboard (front-end asks them to login if they are not already)
- they accept/reject their invites in front-end
- POST api/invite/respond authenticates the already logged in user

What's the "supabase way" to accomplish this?  

How do I find the user by their email without paging through all the users?

Thank you!

**Comment by GaryAustin1** [comment]
Not sure I follow your user flows, but you can have an postgres function (security definer type) select auth.users table where email = xxxx and return their id.  This could be called by an rpc call from the REST API's.   But you might want to remove grants on it from anon and authenticated users depending where you are using it.  You normally don't want to allow probing of emails to the general public.

**Reply by MichaelJCole** [reply]
Hi Gary, thanks for the reply.  I''m new to supabase and using it with nuxt, so I'm still trying to grok your response.   The query to check if a user exists would run in a server-side api call.  So browser -> node -> supabase.

I'm building an invite system where users can invite each other to their projects - but realizing I can't query auth.users.

Do you know of an existing invite design I could use?  I need to send a custom email.  supabase-auth's email templates aren't customizable enough.

I want to get the user emails from this table - maybe I should just add the email fields to the table so I don't have to join to get them:

"CREATE TABLE public.invites (expires_at timestamp with time zone, id uuid NOT NULL, workspace_id uuid NOT NULL, user_id uuid NOT NULL, created_at timestamp with time zone, updated_at timestamp with time zone, created_by uuid NOT NULL, status text NOT NULL, message text);"

Am I missing some documentation relevant to this?  Thanks for the reply,
